diesel/src/pg/types/ranges.rs

```diff
@@ -0,0 +1,125 @@
+
+use byteorder::{ReadBytesExt, WriteBytesExt, NetworkEndian};
+use backend::Debug;
+use pg::{Pg, PgTypeMetadata};
+use types::*;
+use std::collections::Bound;
+use std::error::Error;
+use query_source::Queryable;
+
+bitflags! {
+    struct RangeBound: u8 {
+        const RANGE_EMPTY = 0x01;
```

Comment:
I don't feel super strongly either way, I just find the binary version easier to read.